rts/motoko-rts/src/idl.rs

```diff
@@ -1221,27 +1239,27 @@ unsafe fn idl_alloc_typtbl<M: Memory>(
     type_offsets: Value,
     typtbl_out: *mut *mut *mut u8,
     typtbl_end_out: *mut *mut u8,
-    typtbl_size_out: *mut u32,
+    typtbl_size_out: *mut usize,
 ) {
     let mut type_descriptor = TypeDescriptor::new(candid_data, type_offsets);
     *typtbl_out = type_descriptor.build_type_table(mem);
     *typtbl_end_out = type_descriptor.type_table_end();
-    *typtbl_size_out = type_descriptor.type_count() as u32;
+    *typtbl_size_out = type_descriptor.type_count();
 }
 
 #[no_mangle]
 unsafe extern "C" fn idl_sub(
-    rel_buf: *mut u32, // a buffer with at least 2 * typtbl_size1 * typtbl_size2 bits
+    rel_buf: *mut usize, // a buffer with at least 2 * typtbl_size1 * typtbl_size2 bits
     typtbl1: *mut *mut u8,
     typtbl2: *mut *mut u8,
     typtbl_end1: *mut u8,
     typtbl_end2: *mut u8,
-    typtbl_size1: u32,
-    typtbl_size2: u32,
+    typtbl_size1: usize,
+    typtbl_size2: usize,
```

Comment:
I guess these _sizes could remain u32, given they come from the size of a candid type table.

Reply:
Yes, in practice, should remain `u32`. 
Internally, I believe it is derived for the type table blob size and , which is `usize`. 
I'll add this to the backlog for later consideration of `u32` downgrading.

Reply:
Oh, I assumed it was the number of entries in the type table, not the byte-count.